contracts/upgradeable_contracts/ChaiConnector.sol

```diff
@@ -0,0 +1,269 @@
+pragma solidity 0.4.24;
+
+import "../interfaces/IChai.sol";
+import "../interfaces/ERC677Receiver.sol";
+import "./Ownable.sol";
+import "./ERC20Bridge.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+* @title ChaiConnector
+* @dev This logic allows to use Chai token (https://github.com/dapphub/chai)
+*/
+contract ChaiConnector is Ownable, ERC20Bridge {
+    using SafeMath for uint256;
+
+    bytes32 internal constant CHAI_TOKEN_ENABLED = 0x2ae87563606f93f71ad2adf4d62661ccdfb63f3f508f94700934d5877fb92278; // keccak256(abi.encodePacked("chaiTokenEnabled"))
+    bytes32 internal constant INTEREST_RECEIVER = 0xd88509eb1a8da5d5a2fc7b9bad1c72874c9818c788e81d0bc46b29bfaa83adf6; // keccak256(abi.encodePacked("interestReceiver"))
+    bytes32 internal constant INTEREST_COLLECTION_PERIOD = 0x68a6a652d193e5d6439c4309583048050a11a4cfb263a220f4cd798c61c3ad6e; // keccak256(abi.encodePacked("interestCollectionPeriod"))
+    bytes32 internal constant LAST_TIME_INTEREST_PAYED = 0x120db89f168bb39d737b6a1d240da847e2ead5ecca5b2e4c5e94edbe39d614d9; // keccak256(abi.encodePacked("lastTimeInterestPayed"))
+    bytes32 internal constant INVESTED_AMOUNT = 0xb6afb3323c9d7dc0e9dab5d34c3a1d1ae7739d2224c048d4ee7675d3c759dd1b; // keccak256(abi.encodePacked("investedAmount"))
+    bytes32 internal constant MIN_DAI_TOKEN_BALANCE = 0xce70e1dac97909c26a87aa4ada3d490673a153b3a75b22ea3364c4c7df7c551f; // keccak256(abi.encodePacked("minDaiTokenBalance"))
+
+    uint256 internal constant ONE = 10**27;
+
+    /**
+    * @dev Throws if chai token is not enabled
+    */
+    modifier chaiTokenEnabled {
+        require(isChaiTokenEnabled());
+        _;
```

Comment:
Let's add the following line, as in the other cases,  before `_;` :
```
/* solcov ignore next */
```
so the coverage tool doesn't incorrectly interpret it as an uncovered line